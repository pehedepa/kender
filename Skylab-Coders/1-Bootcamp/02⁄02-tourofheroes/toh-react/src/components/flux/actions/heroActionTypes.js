const heroActionTypes = {
  ADD_HERO: 'ADD_HERO',
  DELETE_HERO: 'DELETE_HERO',
  UPDATE_HERO: 'UPDATE HERO',
  LOAD_HEROES: 'LOAD HEROES',
};

export default heroActionTypes;
