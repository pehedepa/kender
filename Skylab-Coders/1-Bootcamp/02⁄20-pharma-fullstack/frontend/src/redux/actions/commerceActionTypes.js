export default {
  LOAD_ITEMS: 'LOAD_ITEMS',
  ADD_ITEM_TO_CART: 'ADD_ITEM_TO_CART'
};
