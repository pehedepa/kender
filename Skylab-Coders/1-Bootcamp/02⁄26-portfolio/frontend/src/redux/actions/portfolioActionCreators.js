import axios from 'axios';
import portfolioActionTypes from './portfolioActionTypes';

const api = 'http://localhost:5000/api/portfolio';

export default function getGenericData() {
  const generic = '/generic';
  return async (dispatch) => {
    const { data } = await axios(api + generic);
    dispatch({
      type: portfolioActionTypes.GET_GENERIC,
      data
    });
  };
}

export function getTradeSkillsData() {
  const tradeSkills = '/tradeskills';
  return async (dispatch) => {
    const { data } = await axios(api + tradeSkills);
    dispatch({
      type: portfolioActionTypes.GET_TRADESKILLS,
      data
    });
  };
}
