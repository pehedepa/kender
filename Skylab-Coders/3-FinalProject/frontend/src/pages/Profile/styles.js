import { StyleSheet } from 'react-native'
import { ThemeVars } from '../../styles/theme.styles'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    padding: ThemeVars.CONTAINER_PADDING
  },
  flatlistContainer: {
    paddingVertical: 5,
    paddingHorizontal: 20
  },
  avatar: {
    margin: 5,
    width: 150,
    height: 150,
    borderRadius: 100,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5
  },
  icon: {
    maxHeight: 25,
    resizeMode: 'contain'
  },
  subTitle: {
    color: ThemeVars.DARK_COLOR,
    fontFamily: ThemeVars.FONT_FAMILY_MEDIUM,
    fontSize: 14,
    marginVertical: 10
  },
  input: {
    fontFamily: ThemeVars.FONT_FAMILY_MEDIUM,
    borderRadius: 5,
    marginBottom: 5,
    width: '100%',
    paddingVertical: 10,
    paddingHorizontal: 15,
    borderColor: ThemeVars.BORDER_COLOR,
    borderWidth: 1,
    backgroundColor: '#fff'
  },
  inputIconWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    fontFamily: ThemeVars.FONT_FAMILY_MEDIUM,
    borderRadius: 5,
    marginBottom: 5,
    width: '100%',
    paddingVertical: 10,
    paddingHorizontal: 15,
    borderColor: ThemeVars.BORDER_COLOR,
    borderWidth: 1,
    backgroundColor: '#fff'
  },
  card: {
    backgroundColor: '#ffffff',
    borderRadius: 2,
    width: '100%',
    padding: 15,
    alignItems: 'center',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    borderLeftWidth: 4,
    borderColor: ThemeVars.PRIMARY_COLOR
  },
  text: {
    fontFamily: ThemeVars.FONT_FAMILY_LIGHT,
    paddingLeft: 10
  }
})

export default styles
