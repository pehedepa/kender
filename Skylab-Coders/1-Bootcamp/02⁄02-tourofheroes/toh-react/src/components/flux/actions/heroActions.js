import dispatcher from '../dispatcher/dispatcher';
import heroActionTypes from './heroActionTypes';

export function saveHero(hero) {
  debugger;
  const action = {
    type: hero.id ? heroActionTypes.UPDATE_HERO : heroActionTypes.ADD_HERO,
    hero,
  };

  dispatcher.dispatch(action);
}

export function deleteHero({ id }) {
  debugger;
  const action = {
    type: heroActionTypes.DELETE_HERO,
    heroId: id,
  };

  dispatcher.dispatch(action);
}

export function loadHero(heroId) {
  const action = {
    type: heroActionTypes.LOAD_HERO,
    heroId,
  };

  dispatcher.dispatch(action);
}
