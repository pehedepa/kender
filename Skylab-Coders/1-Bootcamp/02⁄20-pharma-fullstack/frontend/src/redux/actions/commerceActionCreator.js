import axios from 'axios';
import commerceActionTypes from './commerceActionTypes';

const api = 'http://localhost:5000/api/commerce';

export default function getItems() {
  return async (dispatch) => {
    const { data } = await axios(api);
    dispatch({
      type: commerceActionTypes.LOAD_ITEMS,
      data
    });
  };
}

export function pushItemToCart(name, _id, url, stock, price) {
  const payload = {
    name,
    _id,
    url,
    stock,
    price
  };

  return (dispatch) => {
    dispatch({
      type: commerceActionTypes.ADD_ITEM_TO_CART,
      data: payload
    });
  };
}
