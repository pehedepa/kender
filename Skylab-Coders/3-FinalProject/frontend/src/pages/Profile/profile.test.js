import React from 'react'
import Profile from './index'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import * as actions from '../../redux/actions/actionCreators'
import { render, fireEvent } from '@testing-library/react-native'
import thunk from 'redux-thunk'

describe('Given a CreateAd function', () => {
  const mockStore = configureStore([thunk])
  let store
  let navigation

  beforeEach(() => {
    jest.spyOn(actions, 'uploadProfile').mockReturnValueOnce({ type: '' })
    store = mockStore({
      userObject: {
        _id: '50',
        name: 'pablo',
        email: 'kender@gmail.com',
        address: 'carrer',
        cc: '5555',
        location: 'barcelona'
      },
      productList: [{ user: { _id: '50' } }]
    })
    navigation = {
      navigate: jest.fn(),
      replace: jest.fn()
    }
  })

  describe('When we invoke Profile function', () => {
    test('Then the render should match its snaptshot', () => {
      const tree = render(<Provider store={store}><Profile></Profile></Provider>)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('When we press Update Button all textInputs with string values', () => {
    test('Then the render should match its snaptshot', () => {
      const tree = render(<Provider store={store}><Profile navigation={navigation}></Profile></Provider>)

      const name = tree.getByPlaceholderText('pablo')
      fireEvent.changeText(name, { text: 'manolo' })

      const address = tree.getByPlaceholderText('carrer')
      fireEvent.changeText(address, { text: 'unaltrecarrer' })

      const location = tree.getByPlaceholderText('barcelona')
      fireEvent.changeText(location, { text: 'madrid' })

      const cc = tree.getByPlaceholderText('5555')
      fireEvent.changeText(cc, { text: '55544' })

      expect(tree).toMatchSnapshot()
    })
  })

  describe('When userObject lacks location/cc/photo/name', () => {
    test('Then those useEffects should start with an empty string as value', () => {
      store = mockStore({
        userObject: {
          _id: '50',
          name: '',
          email: 'kender@gmail.com',
          address: '',
          cc: '',
          location: ''
        },
        productList: [{ user: { _id: '50' } }]
      })

      const tree = render(<Provider store={store}><Profile navigation={navigation}></Profile></Provider>)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('When Update Profile button is pressed', () => {
    test('Then updateUserData function should be invoked', () => {
      const tree = render(<Provider store={store}><Profile navigation={navigation}></Profile></Provider>)
      const button = tree.getByTestId('update')
      fireEvent.press(button)
      expect(actions.uploadProfile).toMatchSnapshot()
    })
  })
})
