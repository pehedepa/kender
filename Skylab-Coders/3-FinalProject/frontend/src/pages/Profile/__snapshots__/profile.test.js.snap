// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CreateAd function When Update Profile button is pressed Then updateUserData function should be invoked 1`] = `[MockFunction]`;

exports[`Given a CreateAd function When userObject lacks location/cc/photo/name Then those useEffects should start with an empty string as value 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "padding": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratLight",
        "fontSize": 22,
        "marginVertical": 10,
      }
    }
  >
    Profile Settings
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "borderRadius": 100,
          "elevation": 5,
          "height": 150,
          "margin": 5,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 150,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 152, 143, 1)",
            "borderColor": "#e0e0e0",
            "borderRadius": 200,
            "elevation": 5,
            "height": 150,
            "padding": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 150,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="name"
    maxLength={30}
    onChangeText={[Function]}
    placeholder=""
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="street-address"
    maxLength={30}
    onChangeText={[Function]}
    placeholder="Write down your full address"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="sentences"
    autoCompleteType="off"
    maxLength={25}
    onChangeText={[Function]}
    placeholder="Write down your location"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="cc-number"
      keyboardType="number-pad"
      maxLength={24}
      onChangeText={[Function]}
      placeholder="Insert your Credit Card details"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "paddingRight": 8,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "maxHeight": 25,
              "resizeMode": "contain",
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratMedium",
        "fontSize": 14,
        "marginVertical": 10,
      }
    }
  >
    PRODUCTS LISTED:
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "user": Object {
            "_id": "50",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ffffff",
                      "borderColor": "#FF988F",
                      "borderLeftWidth": 4,
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "padding": 15,
                      "width": "100%",
                    },
                    Object {
                      "elevation": 3,
                      "shadowColor": "#2F2C23",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                  ]
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "maxHeight": 25,
                      "resizeMode": "contain",
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontFamily": "MontserratLight",
                      "paddingLeft": 10,
                    }
                  }
                />
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FF988F",
              },
            ]
          }
        >
          Update Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Given a CreateAd function When we invoke Profile function Then the render should match its snaptshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "padding": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratLight",
        "fontSize": 22,
        "marginVertical": 10,
      }
    }
  >
    Profile Settings
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "borderRadius": 100,
          "elevation": 5,
          "height": 150,
          "margin": 5,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 150,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 152, 143, 1)",
            "borderColor": "#e0e0e0",
            "borderRadius": 200,
            "elevation": 5,
            "height": 150,
            "padding": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 150,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="name"
    maxLength={30}
    onChangeText={[Function]}
    placeholder="pablo"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value="pablo"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="street-address"
    maxLength={30}
    onChangeText={[Function]}
    placeholder="carrer"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value="carrer"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="sentences"
    autoCompleteType="off"
    maxLength={25}
    onChangeText={[Function]}
    placeholder="barcelona"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value="barcelona"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="cc-number"
      keyboardType="number-pad"
      maxLength={24}
      onChangeText={[Function]}
      placeholder="5555"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "paddingRight": 8,
        }
      }
      underlineColorAndroid="transparent"
      value="5555"
    />
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "maxHeight": 25,
              "resizeMode": "contain",
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratMedium",
        "fontSize": 14,
        "marginVertical": 10,
      }
    }
  >
    PRODUCTS LISTED:
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "user": Object {
            "_id": "50",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ffffff",
                      "borderColor": "#FF988F",
                      "borderLeftWidth": 4,
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "padding": 15,
                      "width": "100%",
                    },
                    Object {
                      "elevation": 3,
                      "shadowColor": "#2F2C23",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                  ]
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "maxHeight": 25,
                      "resizeMode": "contain",
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontFamily": "MontserratLight",
                      "paddingLeft": 10,
                    }
                  }
                />
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FF988F",
              },
            ]
          }
        >
          Update Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Given a CreateAd function When we press Update Button all textInputs with string values Then the render should match its snaptshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "padding": 25,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratLight",
        "fontSize": 22,
        "marginVertical": 10,
      }
    }
  >
    Profile Settings
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "borderRadius": 100,
          "elevation": 5,
          "height": 150,
          "margin": 5,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 150,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 152, 143, 1)",
            "borderColor": "#e0e0e0",
            "borderRadius": 200,
            "elevation": 5,
            "height": 150,
            "padding": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 150,
          }
        }
      />
    </View>
  </RNGestureHandlerButton>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="name"
    maxLength={30}
    onChangeText={[Function]}
    placeholder="pablo"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value={
      Object {
        "text": "manolo",
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="words"
    autoCompleteType="street-address"
    maxLength={30}
    onChangeText={[Function]}
    placeholder="carrer"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value={
      Object {
        "text": "unaltrecarrer",
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="sentences"
    autoCompleteType="off"
    maxLength={25}
    onChangeText={[Function]}
    placeholder="barcelona"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value={
      Object {
        "text": "madrid",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#e0e0e0",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontFamily": "MontserratMedium",
        "marginBottom": 5,
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCompleteType="cc-number"
      keyboardType="number-pad"
      maxLength={24}
      onChangeText={[Function]}
      placeholder="5555"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "flex": 1,
          "paddingRight": 8,
        }
      }
      underlineColorAndroid="transparent"
      value={
        Object {
          "text": "55544",
        }
      }
    />
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "maxHeight": 25,
              "resizeMode": "contain",
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text
    style={
      Object {
        "color": "#2F2C23",
        "fontFamily": "MontserratMedium",
        "fontSize": 14,
        "marginVertical": 10,
      }
    }
  >
    PRODUCTS LISTED:
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "user": Object {
            "_id": "50",
          },
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                  "paddingHorizontal": 20,
                  "paddingVertical": 5,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ffffff",
                      "borderColor": "#FF988F",
                      "borderLeftWidth": 4,
                      "borderRadius": 2,
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "padding": 15,
                      "width": "100%",
                    },
                    Object {
                      "elevation": 3,
                      "shadowColor": "#2F2C23",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                  ]
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "maxHeight": 25,
                      "resizeMode": "contain",
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontFamily": "MontserratLight",
                      "paddingLeft": 10,
                    }
                  }
                />
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#FF988F",
              },
            ]
          }
        >
          Update Profile
        </Text>
      </View>
    </View>
  </View>
</View>
`;
